Gem/Code/Source/ProceduralSkinnedMeshUtils.cpp

```diff
@@ -62,12 +98,52 @@ namespace AtomSampleViewer
-        auto bitangentBuffer = CreateBufferAsset(proceduralMesh.m_bitangents.data(), proceduralMesh.m_bitangents.size(), AZ::RHI::Format::R32G32B32_FLOAT);
-        auto uvBuffer = CreateBufferAsset(proceduralMesh.m_uvs.data(), proceduralMesh.m_uvs.size(), AZ::RHI::Format::R32G32_FLOAT);
+        uint32_t submeshCount = proceduralMesh.GetSubMeshCount();
+        uint32_t verticesPerSubmesh = aznumeric_caster(proceduralMesh.m_positions.size());
+        uint32_t totalVertices = verticesPerSubmesh * submeshCount;
+
+        uint32_t jointIdCountPerSubmesh = verticesPerSubmesh * proceduralMesh.GetInfluencesPerVertex();
+        uint32_t extraJointIdCount = AZ::RPI::CalculateJointIdPaddingCount(jointIdCountPerSubmesh);
+        uint32_t extraPackedIdCount = extraJointIdCount / 2;
+
+        // Copy the original buffer data n-times to create the data for extra sub-meshes
+        DuplicateVertices(proceduralMesh.m_indices, aznumeric_caster(proceduralMesh.m_indices.size()), submeshCount);
+        DuplicateVertices(proceduralMesh.m_positions, verticesPerSubmesh, submeshCount);
+        DuplicateVertices(proceduralMesh.m_normals, verticesPerSubmesh, submeshCount);
+        DuplicateVertices(proceduralMesh.m_tangents, verticesPerSubmesh, submeshCount);
+        DuplicateVertices(proceduralMesh.m_bitangents, verticesPerSubmesh, submeshCount);
+        DuplicateVertices(proceduralMesh.m_uvs, verticesPerSubmesh, submeshCount);
+        DuplicateVertices(proceduralMesh.m_blendWeights, verticesPerSubmesh * proceduralMesh.GetInfluencesPerVertex(), submeshCount);
+
+        // Insert the jointId padding first before duplicating
+        AZStd::vector<uint32_t> extraIds(extraPackedIdCount, 0);
+
+        // Track the count of 32-byte 'elements' (packed) and offsets for creating sub-mesh views
+        uint32_t jointIdElementCountPerSubmesh = aznumeric_caster(proceduralMesh.m_blendIndices.size());
+        uint32_t jointIdOffsetElementsPerSubmesh = jointIdElementCountPerSubmesh + extraPackedIdCount;
+
+        proceduralMesh.m_blendIndices.insert(proceduralMesh.m_blendIndices.end(), extraIds.begin(), extraIds.end());
+        DuplicateVertices(
+            proceduralMesh.m_blendIndices, aznumeric_caster(proceduralMesh.m_blendIndices.size()), submeshCount);
+
+        // Offset duplicate positions in the +y direction, so each sub-mesh ends up in a unique position
+        for (uint32_t subMeshIndex = 1; subMeshIndex < submeshCount; ++subMeshIndex)
+        {
+            for (uint32_t i = 0; i < verticesPerSubmesh; ++i)
+            {
+                proceduralMesh.m_positions[subMeshIndex*verticesPerSubmesh + i][1] +=
+                    aznumeric_cast<float>(subMeshIndex) * proceduralMesh.GetSubMeshYOffset();
+            }
+        }
+
+        auto indexBuffer = CreateTypedBufferAsset(proceduralMesh.m_indices.data(), proceduralMesh.m_indices.size(), AZ::RHI::Format::R32_FLOAT);
```

Comment:
Questions:
1. Why is the index buffer format R32_Float? it shouldn't matter but isn't it supposed to be R32_UINT?
2. What happens if the original format is 16 bits, how is that supported (probably not here where it's only the buffer creation) with the right space required?
3. Isn't there a way to simply take this from the original mesh based on its buffer descriptor?